reco_utils/dataset/criteo.py

```diff
@@ -0,0 +1,188 @@
+    http://labs.criteo.com/2013/12/download-terabyte-click-logs/ 
+
+    Args:
+        spark (pySpark.SparkSession): Spark session.
+        size (str): Dataset size. It can be "sample" or "full".
+        local_cache_path (str): Path where to cache the tar.gz file locally.
+        header (list): Dataset header names.
+        dbfs_datapath (str): Where to store the extracted files on Databricks.
+        dbutils (Databricks.dbutils): Databricks utility object.
+  
+    Returns:
+        pySpark.DataFrame: Criteo DAC training dataset.
+    """
+    with _real_path(local_cache_path) as path:
+        filepath = download_criteo(size, path)
+        filepath = extract_criteo(size, filepath)
+
+        if is_databricks():
+            try:
+                # Driver node's file path
+                node_path = "file:" + filepath
+                ## needs to be on dbfs to load
+                dbutils.fs.cp(node_path, dbfs_datapath, recurse=True)
+                path = dbfs_datapath
+            except:
+                raise ValueError(
+                    "To use on a Databricks notebook, dbutils object should be passed as an argument"
+                )
+        else:
+            path = filepath
+
+        schema = _get_spark_schema(header)
+        df = spark.read.csv(path, schema=schema, sep="\t", header=False)
+    return df
+
+
+def download_criteo(size="sample", work_directory="."):
+    """Download criteo dataset as a compressed file.
+
+    Args:
+        size (str): Size of criteo dataset. It can be "full" or "sample".
+        work_directory (str): Working directory.
+
+    Returns:
+        str: Path of the downloaded file.
+
+    """
+    url = CRITEO_URL[size]
+    return maybe_download(url, work_directory=work_directory)
+
+
+def extract_criteo(size, compressed_file, path=None):
+    """Extract Criteo dataset tar.
+
+    Args:
+        size (str): Size of criteo dataset. It can be "full" or "sample".
+        compressed_file (str): Path to compressed file.
+        path (str): Path to extract the file.
+    
+    Returns:
+        str: Path to the extracted file.
+    
+    """
+    if path is None:
+        folder = os.path.dirname(compressed_file)
+        extracted_dir = os.path.join(folder, "dac")
+    else:
+        extracted_dir = path
+
+    with tarfile.open(compressed_file) as tar:
+        tar.extractall(extracted_dir)
+
+    filename_selector = {
+        "sample": "dac_sample.txt",
+        "full": "train.txt"
+    }
+    return os.path.join(extracted_dir, filename_selector[size])
+
+
+def _get_spark_schema(header):
+    ## create schema
+    schema = StructType()
+    ## do label + ints
+    n_ints = 14
+    for i in range(n_ints):
+        schema.add(StructField(header[i], IntegerType()))
+    ## do categoricals
+    for i in range(26):
+        schema.add(StructField(header[i + n_ints], StringType()))
+    return schema
+
+
+@contextmanager
+def _real_path(path):
```

Comment:
@loomlike we should do DRY here. However, this code is a little bit different to what you have in movielens. I'm not sure how to homogenize

Reply:
I got it. What about this:
1. Put this `real_path` to url_utils (btw we talked about renaming this to dataset_utils, right?)
2. Use that function from criteo.py
3. Refactor movielens to use the same function. Probablly, to check "zip" file name from outside of this function. - either you or I can refactor this.

Reply:
this looks reasonable to me

Reply:
Sounds good. @loomlike if you are free today, feel free to push directly to the branch. If not I'll do it tomorrow

Reply:
@loomlike I added an issue for the renaming https://github.com/Microsoft/Recommenders/issues/662